### PR TITLE
Save switches states in shared preferences

### PR DESCRIPTION
**Pull request by balintvecsey**

### CODE REVIEW

File: app/src/main/java/com/greenfox/opal/gitinder/fragments/SettingsFragment.java [hunk 1]
```diff
@@ -1,22 +1,72 @@
...
 
 public class SettingsFragment extends Fragment {
 
   private static final String TAG = "SettingsFragment";
 
+  Switch switchNotifications;
+  Switch switchSync;
+  @Inject
+  SharedPreferences preferences;
+  SharedPreferences.Editor editor;
```
Comment (adamgyulavari) [inline app/src/main/java/com/greenfox/opal/gitinder/fragments/SettingsFragment.java]: @balintvecsey order of fields should be constants, dependencies, views, other fields

File: app/src/main/java/com/greenfox/opal/gitinder/fragments/SettingsFragment.java [hunk 2]
```diff
@@ -1,22 +1,72 @@
...
+          editor.putBoolean(getString(R.string.settings_notifications), true);
+          editor.apply();
+          Toast.makeText(getContext(), getString(R.string.settings_notifications_on), Toast.LENGTH_SHORT).show();
+        } else {
+          editor.putBoolean(getString(R.string.settings_notifications), false);
+          editor.apply();
+          Toast.makeText(getContext(), getString(R.string.settings_notifications_off), Toast.LENGTH_SHORT).show();
+        }
+      }
+    });
```
Comment (adamgyulavari) [inline app/src/main/java/com/greenfox/opal/gitinder/fragments/SettingsFragment.java:51]: @balintvecsey also you dont use actually the switches other than here, so they could be local variables

File: app/src/main/java/com/greenfox/opal/gitinder/fragments/SwipingFragment.java [hunk 3]
```diff
@@ -85,19 +98,49 @@ public void onScroll(float v) {
...
+      public void onClick(View v) {
+        if (!adapter.isEmpty()) {
+          Log.d("dev", Direction.RIGHT.toString());
+          onSwipingRequest(preferences.getString("X-GiTinder-token", "abcd1234"), adapter.getItem(0).getLogin(), Direction.RIGHT);
+          adapter.remove(adapter.getItem(0));
+          adapter.notifyDataSetChanged();
+          flingAdapterView.removeAllViewsInLayout();
+        }
+      }
+    });
```
Comment (adamgyulavari) [inline app/src/main/java/com/greenfox/opal/gitinder/fragments/SwipingFragment.java]: @balintvecsey why is these changes here?

**Review by adamgyulavari — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/greenfox/opal/gitinder/fragments/SwipingFragment.java (hunk 3 above)
Comment (balintvecsey) [reply app/src/main/java/com/greenfox/opal/gitinder/fragments/SwipingFragment.java]: I made this branch from balint-connected-swipe-request-to-buttons

File: app/src/test/java/com/greenfox/opal/gitinder/SettingsFragmentTest.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+  public void defaultNotificationSwitchState() {
+    Switch switchNotification = (Switch) settingsFragment.getView().findViewById(R.id.switch_notifications);
+    assertFalse(switchNotification.isChecked());
+  }
+
+  @Test
+  public void saveNotificationSwitchStateInSharedPreferences() {
+    Switch switchNotification = (Switch) settingsFragment.getView().findViewById(R.id.switch_notifications);
+    switchNotification.performClick();
+    assertTrue(sharedPreferences.contains("Enable Notifications"));
```
Comment (adamgyulavari) [inline app/src/test/java/com/greenfox/opal/gitinder/SettingsFragmentTest.java]: @balintvecsey you should test the value of it as well

File: app/src/main/java/com/greenfox/opal/gitinder/fragments/SettingsFragment.java [hunk 5]
```diff
@@ -1,22 +1,70 @@
...
+    final View view = inflater.inflate(R.layout.fragment_settings, container, false);
+    final Switch switchNotifications = (Switch) view.findViewById(R.id.switch_notifications);
+    final Switch switchSync = (Switch) view.findViewById(R.id.switch_sync);
+    editor = getActivity().getSharedPreferences(getString(R.string.switch_state), Context.MODE_PRIVATE).edit();
+
+    switchNotifications.setOnClickListener(new OnClickListener() {
+      @Override
+      public void onClick(View v) {
+        if(switchNotifications.isChecked()) {
+          editor.putBoolean(getString(R.string.settings_notifications), true);
```
Comment (adamgyulavari) [inline app/src/main/java/com/greenfox/opal/gitinder/fragments/SettingsFragment.java]: @balintvecsey these string keys doesnt have to be string resources, they are perfectly fine as constants like `TAG`

Comment (adamgyulavari) [reply app/src/main/java/com/greenfox/opal/gitinder/fragments/SettingsFragment.java]: that way you can use them in the tests as well

**Review by adamgyulavari — CHANGES_REQUESTED** [review]

**Pull request merged**